packages/runtime-core/src/hydrationStrategies.ts

```diff
@@ -37,7 +47,14 @@ export const hydrateOnVisible: HydrationStrategyFactory<
       break
     }
   }, opts)
-  forEach(el => ob.observe(el))
+  forEach(el => {
+    if (elementIsVisibleInViewport(el)) {
+      hydrate()
+      ob.disconnect()
+      return () => {}
```

Comment:
After looking at the benchmark I think this is worthwhile, but the implementation needs some adjustments.

Right now there is no way to break out of the `forEach` loop, returning an empty function here doesn't do anything. We need to refactor `forEach` so that the loop breaks if the callback returns `false`.

Reply:
I see. I'll look into refactoring the function, thank you!

Reply:
Is the proposed fix functional?